cuqi/experimental/mcmc/_langevin_algorithm.py

```diff
@@ -30,27 +111,131 @@ def step(self):
+
+    Parameters
+    ----------
+
+    target : `cuqi.distribution.Distribution`
+        The target distribution to sample. Must have logpdf and gradient method. Custom logpdfs 
+        and gradients are supported by using a :class:`cuqi.distribution.UserDefinedDistribution`.
+    
+    initial_point : ndarray
+        Initial parameters. *Optional*
+
+    scale : int
+        The Langevin diffusion discretization time step.
+
+    callback : callable, *Optional*
+        If set this function will be called after every sample.
+        The signature of the callback function is `callback(sample, sample_index)`,
+        where `sample` is the current sample and `sample_index` is the index of the sample.
+        An example is shown in demos/demo31_callback.py.
+
+
+    Example
+    -------
+    .. code-block:: python
+
+        # Parameters
+        dim = 5 # Dimension of distribution
+        mu = np.arange(dim) # Mean of Gaussian
+        std = 1 # standard deviation of Gaussian
+
+        # Logpdf function
+        logpdf_func = lambda x: -1/(std**2)*np.sum((x-mu)**2)
+        gradient_func = lambda x: -2/(std**2)*(x-mu)
+
+        # Define distribution from logpdf as UserDefinedDistribution (sample and gradients also supported)
+        target = cuqi.distribution.UserDefinedDistribution(dim=dim, logpdf_func=logpdf_func,
+            gradient_func=gradient_func)
+
+        # Set up sampler
+        sampler = cuqi.experimental.mcmc.MALANew(target, scale=1/5**2)
+
+        # Sample
+        sampler.sample(2000)
+
+    A Deblur example can be found in demos/demo28_MALA.py
+    """
+
+    def _accept_or_reject(self, x_star, target_eval_star, target_grad_star):
+        """
+        Accepts the proposed state according to a Metropolis step and updates the sampler's state accordingly, i.e., current_point, current_target_eval, and current_target_grad_eval.
+
+        Parameters
+        ----------
+        x_star : 
+            The proposed state
+
+        target_eval_star: 
+            The log likelihood evaluated at x_star
+
+        target_grad_star: 
+            The gradient of log likelihood evaluated at x_star
+
+        Returns
+        -------
+        scaler
+            1 if accepted, 0 otherwise
+        """
         log_target_ratio = target_eval_star - self.current_target_eval
-        log_prop_ratio = self.log_proposal(self.current_point, x_star, target_grad_star) \
-            - self.log_proposal(x_star, self.current_point,  self.current_target_grad_eval)
+        log_prop_ratio = self._log_proposal(self.current_point, x_star, target_grad_star) \
+            - self._log_proposal(x_star, self.current_point,  self.current_target_grad_eval)
         log_alpha = min(0, log_target_ratio + log_prop_ratio)
 
-        # accept/reject
+        # accept/reject with Metropolis
         acc = 0
         log_u = np.log(cuqi.distribution.Uniform(low=0, high=1).sample())
```

Comment:
Thanks for the spot: there is really no need to bother `cuqi.distribution.Uniform` when we simply need a draw from the uniform distribution. It can be easily done with numpy, i.e., `log_u = np.log(np.random.rand())`. I updated this line accordingly